datafusion/common/src/config.rs

```diff
@@ -157,12 +157,10 @@ macro_rules! config_namespace {
                             // $(#[allow(deprecated)])?
                             {
                                 $(let value = $transform(value);)? // Apply transformation if specified
-                                #[allow(deprecated)]
```

Comment:
Hmm I wonder if there was a historical reason these allow deprecations were present; was it anticipating future compatibility? 🤔

Reply:
it may have been needed for code that was subsequently updated but the `allow(deprecated)` annotation was not. That is why I like the `expect(deprecated)` style as then the compiler will tell you when it is no longer actually needed